scripts/wallet-operator-mapper/query-dkg-events.js

```diff
@@ -0,0 +1,414 @@
+          ecdsaWalletID: ethers.ZeroHash,
+          state: 'Unknown',
+          memberCount: 0,
+          operators: []
+        });
+        continue;
+      }
+
+      // Step 2: Query DKG event for this wallet
+      const dkgEvent = await getDkgEventForWallet(walletRegistry, ecdsaWalletID, createdAt, provider);
+
+      if (!dkgEvent) {
+        log.warn(`  No DKG event found`);
+        walletData.push({
+          walletPKH: wallet.walletPublicKeyHash,
+          btcBalance: parseFloat(wallet.walletBitcoinBalance),
+          ecdsaWalletID: ecdsaWalletID,
+          state: 'Live',
+          memberCount: 0,
+          operators: []
+        });
+        continue;
+      }
+
+      // Step 3: Resolve member IDs to operator addresses
+      const operatorAddresses = [];
+
+      for (const memberId of dkgEvent.memberIds) {
+        const address = await getOperatorAddress(sortitionPool, memberId);
+        if (address) {
+          operatorAddresses.push(address);
+        }
+      }
+
+      // Step 4: Map operators to provider info
+      const operatorInfos = operatorAddresses.map(addr => getOperatorInfo(addr, operators.operators));
+
+      const hasDeprecated = operatorInfos.some(op => op.status === 'DISABLE');
+      const hasActive = operatorInfos.some(op => op.status === 'KEEP');
+
+      log.success(`  Found ${operatorInfos.length} operators (${hasDeprecated ? 'HAS DEPRECATED' : 'no deprecated'})`);
+
+      walletData.push({
+        walletPKH: wallet.walletPublicKeyHash,
+        btcBalance: parseFloat(wallet.walletBitcoinBalance),
+        ecdsaWalletID: ecdsaWalletID,
+        state: 'Live', // Simplified - we know from earlier queries
+        memberCount: operatorInfos.length,
+        memberIds: dkgEvent.memberIds,
+        operators: operatorInfos,
+        hasDeprecatedOperator: hasDeprecated,
+        hasActiveOperator: hasActive,
+        needsSweep: hasDeprecated
+      });
+
+    } catch (error) {
+      log.error(`  Failed: ${error.message}`);
+    }
+  }
+
+  const elapsed = ((Date.now() - startTime) / 1000).toFixed(1);
+  log.success(`\nCompleted in ${elapsed}s\n`);
+
+  // Generate summary
+  log.info('Generating summary...');
+
+  const summary = {
+    totalWallets: walletData.length,
+    totalBTC: walletData.reduce((sum, w) => sum + w.btcBalance, 0).toFixed(8),
+    byProvider: {
+      STAKED: { wallets: 0, btc: 0 },
+      P2P: { wallets: 0, btc: 0 },
+      BOAR: { wallets: 0, btc: 0 },
+      NUCO: { wallets: 0, btc: 0 }
+    }
+  };
+
+  // Calculate per-provider totals
+  walletData.forEach(wallet => {
+    if (wallet.hasDeprecatedOperator) {
+      const deprecatedOps = wallet.operators.filter(op => op.status === 'DISABLE');
+      const providers = new Set(deprecatedOps.map(op => op.provider));
+
+      providers.forEach(provider => {
+        if (summary.byProvider[provider]) {
+          summary.byProvider[provider].wallets++;
+          summary.byProvider[provider].btc += wallet.btcBalance;
```

Comment:
What happens if wallet has operators from multiple deprecated providers? Is that possible? If yes, is it going to be counted multiple times?
```suggestion
          summary.byProvider[provider].btc += wallet.btcBalance / providers.size
```